Wrong version of JNA is loaded by relying on Maven transitive dependencies

Exception in thread "main" java.lang.NoSuchMethodError: com.sun.jna.platform.win32.WinNT$PSID.createFieldsOrder(Ljava/lang/String;)Ljava/util/List;
	at com.sun.jna.platform.win32.WinNT$PSID.<clinit>(WinNT.java:354)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at com.sun.proxy.$Proxy0.<clinit>(Unknown Source)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:739)
	at com.sun.jna.Native.loadLibrary(Native.java:415)
	at com.sun.jna.platform.win32.Advapi32.<clinit>(Advapi32.java:70)
	at oshi.software.os.windows.WindowsOperatingSystem.enableDebugPrivilege(WindowsOperatingSystem.java:639)
Disconnected from the target VM, address: '127.0.0.1:56285', transport: 'socket'
	at oshi.software.os.windows.WindowsOperatingSystem.<clinit>(WindowsOperatingSystem.java:109)
	at oshi.SystemInfo.getOperatingSystem(SystemInfo.java:92)
	at org.wikience.chronos.common.monitor.OSAndHWInfoCollector.<clinit>(OSAndHWInfoCollector.java:22)
	at org.wikience.FileStorePerfTest.main(FileStorePerfTest.java:27)

I have removed all JNA dependencies and reimported the projects. 
I have discovered both jna 4.1.0 and jna 4.5.2 appeared after updating, and only jna-platform 4.5.2

Seems like OSHI calls go to jna 4.1.0 since when I delete jna 4.1.0 it throws

Exception in thread "main" java.lang.NoClassDefFoundError: com/sun/jna/Platform
	at oshi.SystemInfo.<clinit>(SystemInfo.java:59)
	at org.wikience.FileStorePerfTest.main(FileStorePerfTest.java:27)
Caused by: java.lang.ClassNotFoundException: com.sun.jna.Platform
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:338)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 2 more

I will try to seek whether any other artifact depends on jna 4.1.0, but I doubt whether I will find any: my first guess is that OSHI is probably the only artifact using jna.

--- a/oshi-core/pom.xml
+++ b/oshi-core/pom.xml
@@ -39,12 +39,21 @@
 		<url>https://github.com/oshi/oshi.git</url>
 		<tag>HEAD</tag>
 	</scm>
-
+	
+	<properties>
+		<jna.version>5.0.0</jna.version>
+	</properties>
+	
 	<dependencies>
 		<dependency>
 			<groupId>net.java.dev.jna</groupId>
+			<artifactId>jna</artifactId>
+			<version>${jna.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>net.java.dev.jna</groupId>
 			<artifactId>jna-platform</artifactId>
-			<version>5.0.0</version>
+			<version>${jna.version}</version>
 		</dependency>
 	</dependencies>
 </project>

